newIDE/app/public/External/Piskel/piskel-main.js

```diff
@@ -124,6 +128,11 @@ document.getElementById('piskel-frame').onload = function() {
   piskelAppHeader.appendChild(cancelButton);
   cancelButton.addEventListener('click', cancelChanges);
 
+  setFolderButton = editorContentDocument.createElement('button');
+  setFolderButton.textContent = 'Set Folder';
+  piskelAppHeader.appendChild(setFolderButton);
+  setFolderButton.addEventListener('click', selectBaseFolderPath);
+
   // Apply header styles
   Object.keys(headerStyle).forEach(function(key) {
     eval(key).style = headerStyle[key];
```

Comment:
I missed this eval the last time. Please never never ever use eval.
It can be a risk of XSS injection attack, break performance, disable any chance of doing static analysis with Flow, disable all linting that could be done and let the door open for subtle bugs. 99.999% of the time when you want to do something with eval, you can do it otherwise.
More information here: https://javascriptweblog.wordpress.com/2010/04/19/how-evil-is-eval/

Could you try to find another way to do it? If you can't do it automatically, applying manually the style is better than using eval. ("explicit is better than implicit" in programming in most situations).

Reply:
Note that the loss in performance or the risk of XSS attacks here are very very low. But eval is hard to read for the code reader :)

Reply:
@4ian thank you taking the time to explain this. I am learning alot from you :)
I will remove the eval